src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -307,16 +307,19 @@ def create_canopy_index(self,
         if dimension is None:
             try:
                 encoder_dimension = self._encoder.dimension
+                if encoder_dimension is None:
+                    raise RuntimeError(
+                        f"The selected encoder {self._encoder.__class__.__name__} does"
+                        f"not support inferring the vectors' dimensionality."
+                    )
+                dimension = encoder_dimension
             except Exception as e:
                 raise RuntimeError(
-                    f"Failed to infer vectors' dimension from encoder due to error: "
-                    f"{e}. Please fix the error or provide the dimension manually"
+                    f"Canopy has failed to infer vectors' dimensionality using the "
+                    f"selected encoder: {self._encoder.__class__.__name__}. You can "
+                    f"provide the dimension manually, try using a different encoder, or"
+                    f" fix the underlying error:\n{e}"
```

Comment:
this part should have system test

Reply:
It has. And you're right - I need to fix that test